subprojects/core/src/main/java/org/gradle/internal/buildevents/BuildExceptionReporter.java

```diff
@@ -176,28 +204,45 @@ private LinePrefixingStyledTextOutput getLinePrefixingStyledTextOutput() {
     }
 
     private void fillInFailureResolution(FailureDetails details) {
+        BufferingStyledTextOutput resolution = details.resolution;
         if (details.failure instanceof FailureResolutionAware) {
-            ((FailureResolutionAware) details.failure).appendResolution(details.resolution, clientMetaData);
-            if (details.resolution.getHasContent()) {
-                details.resolution.append(' ');
+            ((FailureResolutionAware) details.failure).appendResolution(resolution, clientMetaData);
+            if (resolution.getHasContent()) {
+                resolution.append(' ');
             }
         }
         if (details.exceptionStyle == ExceptionStyle.NONE) {
-            details.resolution.text("Run with ");
-            details.resolution.withStyle(UserInput).format("--%s", LoggingCommandLineConverter.STACKTRACE_LONG);
-            details.resolution.text(" option to get the stack trace. ");
+            resolution.text("Run with ");
+            maybeAddBuildScan(resolution);
+            resolution.withStyle(UserInput).format("--%s", LoggingCommandLineConverter.STACKTRACE_LONG);
+            resolution.text(" option to get the stack trace. ");
         }
         if (loggingConfiguration.getLogLevel() != LogLevel.DEBUG) {
-            details.resolution.text("Run with ");
+            resolution.text("Run with ");
+            maybeAddBuildScan(resolution);
```

Comment:
:x: Along with line 216, this prints the "Run with --scan" twice. 

![screen shot 2017-07-10 at 9 33 11 am](https://user-images.githubusercontent.com/51534/28037829-a55d7860-6571-11e7-8404-a0d78eacd2f1.png)